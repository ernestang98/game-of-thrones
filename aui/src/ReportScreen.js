import * as React from 'react';
import Typography from '@mui/material/Typography';
import '@fontsource/roboto/500.css';
import {
    Box,
    Grid,
    Card,
    CardMedia,
    CardContent,
    List,
    ListItem,
    ListItemButton,
    ListItemIcon,
    ListItemText,
} from "@mui/material";
import {Link} from "react-router-dom";
import GroupsIcon from '@mui/icons-material/Groups';
import EmojiPeopleIcon from '@mui/icons-material/EmojiPeople';
import indiv_student from './static/images/indiv_student.jpg';
import group_of_students from './static/images/group_of_students.jpg';

/**
 * render Report screen to allow admin to select between cohort and individual reports
 * @returns {JSX.Element}
 * @constructor
 */
function ReportScreen() {
    return (
        <Box sx={{ my: 5, }}>
            <Typography flex-grow={1} align="center" variant="h3">
                Select Report Category
            </Typography>
            <Box sx={{ flexGrow: 1 }}>
                <Grid 
                container
                spacing={2}
                direction="row"
                justifyContent="space-evenly"
                alignItems="center"
                >
                    <Card
                    sx={{ my: 7 }}
                    >
                        <CardMedia
                        component="img"
                        height="300"
                        src={indiv_student}
                        alt="Individual Student"
                        />
                        <CardContent>
                        <Typography gutterBottom variant="h5" component="div">
                            Individual Students
                        </Typography>
                        <List>
                            <ListItem id="indiv-link" button component={Link}  to="/indiv-report" disablePadding>
                                <ListItemButton>
                                <ListItemIcon>
                                    <EmojiPeopleIcon />
                                </ListItemIcon>
                                <ListItemText primary="Individual Performance Score" />
                                </ListItemButton>
                            </ListItem>
                        </List>
                        
                        </CardContent>
                    </Card>
                    <Card
                    sx={{ my: 7 }}
                    >
                        <CardMedia
                        component="img"
                        height="300"
                        src={group_of_students}
                        alt="Group of Students"
                        />
                        <CardContent>
                        <Typography gutterBottom variant="h5" component="div">
                            Student Cohort
                        </Typography>
                        <List>
                            <ListItem id="cohort-link" button component={Link}  to="/cohort-report" disablePadding>
                                <ListItemButton>
                                <ListItemIcon>
                                    <GroupsIcon />
                                </ListItemIcon>
                                <ListItemText primary="Cohort Performance Score" />
                                </ListItemButton>
                            </ListItem>
                        </List>
                        </CardContent>
                    </Card>
                </Grid>
            </Box>
        </Box>
    );
}

export default ReportScreen;
