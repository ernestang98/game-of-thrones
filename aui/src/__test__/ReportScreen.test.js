import React from 'react';
import {mount, shallow} from 'enzyme';
import ReportScreen from "../ReportScreen";

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import "@testing-library/jest-dom/extend-expect"
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe("Report Main Screen", () => {
    it("render success", () => {
        const wrapper = shallow(<ReportScreen />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('link to individual report screen', () => {
        const wrapper = shallow(<ReportScreen />);
        expect(wrapper
            .find('#indiv-link')
            .props()
            .to
            )
            .toBe('/indiv-report');
    })

    it('includes link to individual report screen', () => {
        const wrapper = shallow(<ReportScreen />);
        expect(wrapper
            .find('#cohort-link')
            .props()
            .to
            )
            .toBe('/cohort-report');
    })
});

