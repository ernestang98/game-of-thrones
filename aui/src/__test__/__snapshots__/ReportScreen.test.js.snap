// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report Main Screen render success 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "my": 5,
    }
  }
>
  <ForwardRef(Typography)
    align="center"
    flex-grow={1}
    variant="h3"
  >
    Select Report Category
  </ForwardRef(Typography)>
  <ForwardRef(Box)
    sx={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <ForwardRef(Grid)
      alignItems="center"
      container={true}
      direction="row"
      justifyContent="space-evenly"
      spacing={2}
    >
      <ForwardRef(Card)
        sx={
          Object {
            "my": 7,
          }
        }
      >
        <ForwardRef(CardMedia)
          alt="Individual Student"
          component="img"
          height="300"
          src={Object {}}
        />
        <ForwardRef(CardContent)>
          <ForwardRef(Typography)
            component="div"
            gutterBottom={true}
            variant="h5"
          >
            Individual Students
          </ForwardRef(Typography)>
          <ForwardRef(List)>
            <ForwardRef(ListItem)
              button={true}
              component={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              disablePadding={true}
              id="indiv-link"
              to="/indiv-report"
            >
              <ForwardRef(ListItemButton)>
                <ForwardRef(ListItemIcon)>
                  <Memo(ForwardRef(EmojiPeopleIcon)) />
                </ForwardRef(ListItemIcon)>
                <ForwardRef(ListItemText)
                  primary="Individual Performance Score"
                />
              </ForwardRef(ListItemButton)>
            </ForwardRef(ListItem)>
          </ForwardRef(List)>
        </ForwardRef(CardContent)>
      </ForwardRef(Card)>
      <ForwardRef(Card)
        sx={
          Object {
            "my": 7,
          }
        }
      >
        <ForwardRef(CardMedia)
          alt="Group of Students"
          component="img"
          height="300"
          src={Object {}}
        />
        <ForwardRef(CardContent)>
          <ForwardRef(Typography)
            component="div"
            gutterBottom={true}
            variant="h5"
          >
            Student Cohort
          </ForwardRef(Typography)>
          <ForwardRef(List)>
            <ForwardRef(ListItem)
              button={true}
              component={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              disablePadding={true}
              id="cohort-link"
              to="/cohort-report"
            >
              <ForwardRef(ListItemButton)>
                <ForwardRef(ListItemIcon)>
                  <Memo(ForwardRef(GroupsIcon)) />
                </ForwardRef(ListItemIcon)>
                <ForwardRef(ListItemText)
                  primary="Cohort Performance Score"
                />
              </ForwardRef(ListItemButton)>
            </ForwardRef(ListItem)>
          </ForwardRef(List)>
        </ForwardRef(CardContent)>
      </ForwardRef(Card)>
    </ForwardRef(Grid)>
  </ForwardRef(Box)>
</ForwardRef(Box)>
`;
